src/manifest/extensions/ExtensionGuards.ts

```diff
@@ -227,3 +228,15 @@ export const isActionShellExtension = (extension: any): extension is KnowledgeSh
 
   return true;
 };
+
+export const isEditorShellExtension = (extension: any): extension is EditorExtension => {
```

Comment:
Can't this all be simplified to a one liner?
`return Object.prototype.hasOwnProperty.call(extension, 'type') && extension.type === ShellExtensionType.EDITOR;`
or even 
`return extension.type === ShellExtensionType.EDITOR;`